generator/src/main/java/com/scottlogic/deg/generator/restrictions/linear/NumericRestrictions.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.generator.restrictions.linear;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+import static com.scottlogic.deg.common.util.Defaults.*;
+
+public class NumericRestrictions extends LinearRestrictions<BigDecimal> {
+
+    public static final Limit<BigDecimal> NUMERIC_MIN_LIMIT = new Limit<>(NUMERIC_MIN, true);
+    public static final Limit<BigDecimal> NUMERIC_MAX_LIMIT = new Limit<>(NUMERIC_MAX, true);
+    public static final NumericConverter CONVERTER = new NumericConverter();
+
+    public NumericRestrictions(Limit<BigDecimal> min, Limit<BigDecimal> max){
+        this(min, max, DEFAULT_NUMERIC_SCALE);
+    }
+
+    public NumericRestrictions(Limit<BigDecimal> min, Limit<BigDecimal> max, int numericScale){
+        this(min, max, new NumericGranularity(numericScale));
+    }
+    public NumericRestrictions(Limit<BigDecimal> min, Limit<BigDecimal> max, Granularity<BigDecimal> granularity){
+        super(capMin(min), capMax(max), granularity, CONVERTER);
+    }
+
+    private static Limit<BigDecimal> capMax(Limit<BigDecimal> max) {
+        if (max.isAfter(NUMERIC_MAX)) {
+            return new Limit<>(NUMERIC_MAX, true);
+        } else if (max.isBefore(NUMERIC_MIN)) {
```

Comment:
This logic should be the same as in `DateTimeRestrictions`

Reply:
Nice catch! Fixed